monai/transforms/transforms.py

```diff
@@ -214,13 +214,16 @@ def __call__(self, filename):
             img = correct_nifti_header_if_necessary(img)
             header = dict(img.header)
             header['filename_or_obj'] = name
-            header['affine'] = img.affine
+            header['original_affine'] = img.affine.copy()
+            header['affine'] = img.affine.copy()
             header['as_closest_canonical'] = self.as_closest_canonical
+            ndim = img.header['dim'][0]
+            spatial_rank = min(ndim, 3)
```

Comment:
Why only support <= 3 dims? Maybe we need a discussion about dims.

Reply:
it's spatial shape because segmentation output's channel dimension might be different from those of the nifti image input